Import required packages and load the dataset into a variable called data.

In [59]:
import numpy as np
import pandas as pd


In [60]:
data = pd.read_csv("F:/ICT/python_programming/dataset_3/Sales_data_zone_wise_ANOVA.csv")
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [61]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Month     29 non-null     object
 1   Zone - A  29 non-null     int64 
 2   Zone - B  29 non-null     int64 
 3   Zone - C  29 non-null     int64 
 4   Zone - D  29 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.3+ KB


# 1) The mean sales generated by each zone

Next we have to find out the mean sales generated by each zone. To find the mean we can use any of these 2 method.

In [62]:
data.describe()

Zone - A      Zone - B      Zone - C      Zone - D
count  2.900000e+01  2.900000e+01  2.900000e+01  2.900000e+01
mean   1.540493e+06  1.755560e+06  1.772871e+06  1.842927e+06
std    2.619401e+05  1.683899e+05  3.331937e+05  3.750165e+05
min    1.128185e+06  1.527574e+06  1.237722e+06  1.234311e+06
25%    1.305972e+06  1.606010e+06  1.523308e+06  1.520406e+06
50%    1.534390e+06  1.740365e+06  1.767047e+06  1.854412e+06
75%    1.820196e+06  1.875658e+06  2.098463e+06  2.180416e+06
max    2.004480e+06  2.091194e+06  2.290580e+06  2.364132e+06

In [63]:
print('\nMean sales generated by Zone A is: ', round(data['Zone - A'].mean()))
print('\nMean sales generated by Zone B is: ', round(data['Zone - B'].mean()))
print('\nMean sales generated by Zone C is: ', round(data['Zone - C'].mean()))
print('\nMean sales generated by Zone D is: ', round(data['Zone - D'].mean()))


Mean sales generated by Zone A is:  1540493

Mean sales generated by Zone B is:  1755560

Mean sales generated by Zone C is:  1772871

Mean sales generated by Zone D is:  1842927


# 2)Total sales generated by all the zones for each month. 

In [64]:
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [65]:
data['Total_sales'] = data['Zone - A'] + data['Zone - B'] + data['Zone - C'] + data['Zone - D']
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D  Total_sales
0  Month - 1   1483525   1748451   1523308   2267260      7022544
1  Month - 2   1238428   1707421   2212113   1994341      7152303
2  Month - 3   1860771   2091194   1282374   1241600      6475939
3  Month - 4   1871571   1759617   2290580   2252681      8174449
4  Month - 5   1244922   1606010   1818334   1326062      5995328

# 3)Check whether all the zones generate the same amount of sales. Help the company to carry out their study with the help of data provided

Independent variables are also termed as factor. For this experimental design, there is only 1 factor or independent variable to evaluate[Amount of Sales], and therefore, one-way ANOVA method is suitable for analysis.

Null Hypothesis:All the zones generate the same amount of sales

Alternate Hypothesis:All the zones generate the different amount of sales

Significant value, : 0.05 i.e 5%

In [66]:
from scipy.stats import f_oneway

In [67]:
import scipy.stats as stats

In [68]:
fvalue, pvalue = stats.f_oneway(data['Zone - A'], data['Zone - B'], data['Zone - C'], data['Zone - D'])
print(fvalue, pvalue)

5.672056106843581 0.0011827601694503335


From ANOVA test we got f_value = 5.672056106843581 and p_value = 0.0011827601694503335

Insights: The p_value we obtaine dfrom the ANOVA test is smaller than the significant value. Therefore we reject the null hypothesis.
 
Therefore we can conclude that zones generates different amount of sales.